Extract `NumericData` model for tests

kamipo opened the pull request:
Currently `NumericData` model is defined some places.

kamipo committed:
Extract `NumericData` model for tests

Currently `NumericData` model is defined some places.

prathamesh-sonpatki commented on activerecord/test/cases/base_test.rb:
```diff
@@ -908,13 +909,6 @@ def test_default
-    attribute :my_house_population, :integer
-    attribute :atoms_in_universe, :integer
-  end
-
```
Is this not needed now?

prathamesh-sonpatki commented on activerecord/test/cases/calculations_test.rb:
```diff
@@ -17,14 +18,6 @@
-  attribute :world_population, :integer
-  attribute :my_house_population, :integer
-  attribute :atoms_in_universe, :integer
-end
```
Is this also not needed now?

kamipo replied:
Yes, this was added for prevent deprecation message in #15295.
But the deprecation message was removed in #15374.
We do not see deprecation message even if these line are removed.

rafaelfranca commented:
Related with https://github.com/rails/rails/commit/aff73e094640b19cb1bdc975082be7b4767bf4bf. We should put the deprecation back what will require the override of the attributes in the tests.

kamipo committed:
Extract `NumericDataTest` to `test/cases/numeric_data_test.rb`

To ease to find the numeric data tests, extract `NumericDataTest` to
`test/cases/numeric_data_test.rb` dedicated file.

kamipo committed:
Restore the override of numeric attributes properly

`attribute :world_population, :integer` is not a same with default
decimal without scale type unless #26302 is merged.
Should be `attribute :world_population, :big_integer` for now.

kamipo commented:
Restored the override of the attributes properly.
